### PR TITLE
Add from_delayed

### PR DESCRIPTION
**Pull request by phofl**
The expression representation is a bit weird for now, but I'd like to have from_delayed for testing purposes before we implement it properly

ref #525

### CODE REVIEW

File: dask_expr/io/_delayed.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        if self.verify_meta:
+            return (
+                functools.partial(check_meta, meta=self._meta, funcname="from_delayed"),
+                key,
+            )
+        else:
+            return identifiy, key
+
+
+def identifiy(x):
```
Comment (hendrikmakait) [inline dask_expr/io/_delayed.py:76]: Did you mean
```suggestion
def identity(x):
```

File: dask_expr/io/_delayed.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        If True check that the partitions have consistent metadata, defaults to True.
+    """
+    if isinstance(dfs, Delayed) or hasattr(dfs, "key"):
+        dfs = [dfs]
+
+    if len(dfs) == 0:
+        raise TypeError("Must supply at least one delayed object")
+
+    dfs = [
+        delayed(df) if not isinstance(df, Delayed) and hasattr(df, "key") else df
```
Comment (hendrikmakait) [inline dask_expr/io/_delayed.py:115]: I'm not entirely sure what you want to test here, but it feels like you may want to check whether this is an actual or duck-typed delayed which would mirror the check above.
```suggestion
        delayed(df) if not (isinstance(df, Delayed) or hasattr(df, "key")) else df
```

File: dask_expr/io/_delayed.py (hunk 1 above)
Comment (phofl) [reply dask_expr/io/_delayed.py:76]: Oh yes, thx

File: dask_expr/io/_delayed.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    def _filtered_task(self, index: int):
+        key = self.dfs[index]._name
+        if self.verify_meta:
+            return (
+                functools.partial(check_meta, meta=self._meta, funcname="from_delayed"),
+                key,
+            )
+        else:
+            return identifiy, key
```
Comment (hendrikmakait) [inline dask_expr/io/_delayed.py:73]: Sorry, you gotta change this as well.

```suggestion
            return identity, key
```

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl!

**Pull request merged**